Services/UserService/Commands/EditUserCommand.cs

```diff
@@ -0,0 +1,84 @@
+﻿using DTO;
+using DTO.BrokerRequests;
+using DTO.RestRequests;
+using FluentValidation;
+using Kernel;
+using Kernel.CustomExceptions;
+using MassTransit;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+using UserService.Interfaces;
+using UserService.Validators;
+
+namespace UserService.Commands
+{
+    public class EditUserCommand : IEditUserCommand
+    {
+        private readonly IBus busControl;
+        private readonly IValidator<CreateUserRequest> validator;
+        private readonly IValidator<string> validatorForPassword;
+        public EditUserCommand([FromServices] IBus busControl, [FromServices] IValidator<CreateUserRequest> validator,
+            IValidator<string> validatorForPassword)
+        {
+            this.busControl = busControl;
+            this.validator = validator;
+            this.validatorForPassword = validatorForPassword;
+        }
+
+        public async Task<bool> Execute(CreateUserRequest request, string newPassword)
+        {
+            CreateUserRequestValidator createUserRequestValidator = new CreateUserRequestValidator();
+            validator.ValidateAndThrow(request);
+            validatorForPassword.ValidateAndThrow(newPassword);
+
+            string passwordHash = ShaHash.GetPasswordHash(newPassword);
+
+            if (newPassword == null)
+            {
+                throw new NotFoundException("Unable to edit user");
+            }
+            var user = new User
+            {
+                Id = Guid.NewGuid(),
```

Comment:
If this is edit request, why are you setting new Id?

Reply:
Done